internal/controllers/machine/machine_controller_status.go

```diff
@@ -461,15 +507,38 @@ func (c machineConditionCustomMergeStrategy) Merge(conditions []v1beta2condition
 			}
 			// Note: MachineNodeReadyV1Beta2Condition is not relevant for the summary.
 		}
-		return v1beta2conditions.GetDefaultMergePriorityFunc(nil)(condition)
+		return v1beta2conditions.GetDefaultMergePriorityFunc(c.negativePolarityConditionTypes)(condition)
 	}).Merge(conditions, conditionTypes)
 }
 
+func setDeletingCondition(_ context.Context, machine *clusterv1.Machine, reconcileDeleteExecuted bool, deletingReason, deletingMessage string) {
+	if machine.DeletionTimestamp.IsZero() {
+		v1beta2conditions.Set(machine, metav1.Condition{
+			Type:   clusterv1.MachineDeletingV1Beta2Condition,
+			Status: metav1.ConditionFalse,
+			Reason: clusterv1.MachineDeletingDeletionTimestampNotSetV1Beta2Reason,
+		})
```

Comment:
Missing return?

Reply:
Thx!! great catch

Reply:
Btw, it would still have worked because in that case reconcileDeleteExecuted is false (also the reason why the unit test didn't catch it)